holmes/core/tool_calling_llm.py

```diff
@@ -245,22 +255,35 @@ def call(
 
                 for future in concurrent.futures.as_completed(futures):
                     tool_call_result: ToolCallResult = future.result()
-                    tool_calls.append(tool_call_result)
 
-                    tool_response = tool_call_result.result.data
-                    if tool_call_result.result.status == ToolResultStatus.ERROR:
-                        tool_response = f"{tool_call_result.result.error or 'Tool execution failed'}:\n\n{tool_call_result.result.data or ''}".strip()
+                    tool_response_content = (
+                        self._get_formatted_tool_call_response_content(tool_call_result)
+                    )
+                    # TODO: remove this after FE is ready
+                    tool_call_response = tool_call_result.as_dict()
+                    if HOLMES_STRUCTURED_OUTPUT_CONVERSION_FEATURE_FLAG:
+                        tool_call_response["result"] = tool_response_content
+
+                    tool_calls.append(tool_call_response)
```

Comment:
This used to be `tool_call_result` directly before the struct tool result change. Why does it need to change?